packages/smart-wallet/src/smartWallet.js

```diff
@@ -88,13 +88,28 @@ const mapToRecord = map => Object.fromEntries(map.entries());
 /** @typedef {import('./types').RemotePurse} RemotePurse */
 
 /**
+ * @typedef {{
+ * address: string,
+ * bank: ERef<import('@agoric/vats/src/vat-bank').Bank>,
+ * invitationPurse: Purse<'set'>,
+ * }} UniqueParams
+ *
+ * @typedef {{
+ * agoricNames: ERef<import('@agoric/vats').NameHub>,
+ * invitationIssuer: ERef<Issuer<'set'>>,
+ * invitationBrand: Brand<'set'>,
+ * publicMarshaller: Marshaller,
+ * storageNode: ERef<StorageNode>,
+ * zoe: ERef<ZoeService>,
+ * }} SharedParams
+ *
  * @typedef {ImmutableState & MutableState} State
  * - `brandPurses` is precious and closely held. defined as late as possible to reduce its scope.
  * - `offerToInvitationMakers` is precious and closely held.
  * - `brandDescriptors` will be precious. Currently it includes invitation brand and  what we've received from the bank manager.
  * - `purseBalances` is a cache of what we've received from purses. Held so we can publish all balances on change.
  *
- * @typedef {Parameters<initState>[0] & Parameters<initState>[1]} HeldParams
```

Comment:
basing the type of the params was causing a circular reference when I made `initState` to return `State`.